### PR TITLE
Linodecluster requeue

### PR DESCRIPTION
**Pull request by komer3**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: controller/linodecluster_controller.go [hunk 1]
```diff
@@ -117,24 +117,51 @@ func (r *LinodeClusterReconciler) reconcile(
...
 			reterr = err
 		}
 	}()
 
 	// Handle deleted clusters
 	if !clusterScope.LinodeCluster.DeletionTimestamp.IsZero() {
-		return res, r.reconcileDelete(ctx, logger, clusterScope)
+		err := r.reconcileDelete(ctx, logger, clusterScope)
+		if err != nil && !reconciler.HasConditionSeverity(clusterScope.LinodeCluster, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
+			if !reconciler.HasConditionSeverity(clusterScope.LinodeCluster, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
```
Comment (bcm820) [inline controller/linodecluster_controller.go:130]: Only one condition severity check is needed here. Also, we should only requeue when the condition severity is warning, not error (which we consider terminal).

Comment (komer3) [reply controller/linodecluster_controller.go:130]: oops. Thanks for catching that!

File: controller/linodecluster_controller.go [hunk 2]
```diff
@@ -117,24 +117,48 @@ func (r *LinodeClusterReconciler) reconcile(
...
-		if err := r.reconcileCreate(ctx, logger, clusterScope); err != nil {
+		err := r.reconcileCreate(ctx, logger, clusterScope)
+		if err != nil {
+			if !reconciler.HasConditionSeverity(clusterScope.LinodeCluster, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
+				logger.Info("re-queuing cluster/nb creation")
+
+				res = ctrl.Result{RequeueAfter: reconciler.DefaultClusterControllerReconcileDelay}
+
+				return res, nil
+			}
```
Comment (AshleyDumaine) [inline controller/linodecluster_controller.go:161]: should be able to condense this to:
```suggestion
		if err := r.reconcileCreate(ctx, logger, clusterScope); err != nil &&
			!reconciler.HasConditionSeverity(clusterScope.LinodeCluster, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
				logger.Info("re-queuing cluster/nb creation")

				return ctrl.Result{RequeueAfter: reconciler.DefaultClusterControllerReconcileDelay}, nil
			}
```

Comment (AshleyDumaine) [inline controller/linodecluster_controller.go:161]: small tweak
```suggestion
		if err := r.reconcileCreate(ctx, logger, clusterScope); err != nil {
			if !reconciler.HasConditionSeverity(clusterScope.LinodeCluster, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
				logger.Info("re-queuing cluster/nb creation")

				return ctrl.Result{RequeueAfter: reconciler.DefaultClusterControllerReconcileDelay}, nil
			}
```

File: controller/linodecluster_controller.go [hunk 3]
```diff
@@ -117,24 +117,48 @@ func (r *LinodeClusterReconciler) reconcile(
 	// Always close the scope when exiting this function so we can persist any LinodeCluster changes.
 	defer func() {
 		// Filter out any IsNotFound message since client.IgnoreNotFound does not handle aggregate errors
-		if err := clusterScope.Close(ctx); utilerrors.FilterOut(err, apierrors.IsNotFound) != nil && reterr == nil {
+		// if len(clusterScope.LinodeCluster.Finalizers) == 0 {
```
Comment (AshleyDumaine) [inline controller/linodecluster_controller.go:120]: should be safe to remove?

Comment (komer3) [reply controller/linodecluster_controller.go:120]: Oops. Good catch!

File: controller/linodecluster_controller.go [hunk 4]
```diff
@@ -117,24 +117,37 @@ func (r *LinodeClusterReconciler) reconcile(
...
+				return ctrl.Result{RequeueAfter: reconciler.DefaultClusterControllerReconcileDelay}, nil
+			}
+			return res, err
+		}
+		return res, nil
 	}
 
-	if err := clusterScope.AddFinalizer(ctx); err != nil {
+	err := clusterScope.AddFinalizer(ctx)
+	if err != nil {
```
Comment (AshleyDumaine) [inline controller/linodecluster_controller.go:139]:
```suggestion
	if err := clusterScope.AddFinalizer(ctx); err != nil {
```

**Review by AshleyDumaine — APPROVED** [review]

**Review by bcm820 — APPROVED** [review]

**Pull request merged**